flang/lib/Optimizer/Dialect/FIRType.cpp

```diff
@@ -1266,4 +1266,7 @@ void FIROpsDialect::registerTypes() {
       OpenMPPointerLikeModel<fir::LLVMPointerType>>(*getContext());
   fir::LLVMPointerType::attachInterface<
       OpenACCPointerLikeModel<fir::LLVMPointerType>>(*getContext());
+
+  fir::BoxType::attachInterface<OpenMPPointerLikeModel<fir::BoxType>>(
```

Comment:
This doesn't seem right. A box type is an abstract concept that is not "pointer-like" at all. This maybe is a bit more obvious at the LLVM IR level - LLVMTypeConverter::convertBoxTypeAsStruct converts it to a struct not a pointer.

Reply:
Yes, it isn't quite right (although, if we were going to keep it I'd have suggested a different name like OpenMPMapLikeModel to cover a wider range of types), I believe there's a way around adding it if I make the CodeGen pass a little more complex which should be fine, if we did end up going down this route. But I'll investigate the addr_addr of route.